sunpy/conftest.py

```diff
@@ -50,6 +50,14 @@ def pytest_runtest_setup(item):
             pytest.skip(msg.format(item.name))
 
 def pytest_unconfigure(config):
+    tempdir = tempfile.mkdtemp(suffix="_figures")
+    for h in hash.file_list:
+        for name, search_h in hash.hash_library.iteritems():
+            if h == search_h:
```

Comment:
By inspection, it doesn't seem like you need the outer `for` loop.  You could just try to access `file_list` using the hash (e.g., `hash.file_list[search_h]`) and bail out if it fails.